src/components/profile.vue

```diff
@@ -1,354 +1,521 @@
+        this.status.volunteer = "inactive";
+        this.status.profile = "inactive";
+      }
+    },
+    confirmFields: function(field, confirm, label) {
+      if (this.setup) {
+        return (field != confirm) ? `Your ${label} and confirmation do not match`: "";
+      }
+      else { return "" }
+    },
+    confirmMatchBool: function(errMsg) {
+      return (errMsg.length > 0);
+    },
+    confirmAllMatches: function() {
+      return new Promise((resolve, reject) => {
+        this.debouncedConfirmName();
+        this.debouncedConfirmEmail();
+        this.debouncedConfirmPassword();
+        resolve(true);
+      });
+    },
+    debouncedConfirmName: function() {
+      this.confirmErrors.name = this.confirmFields(
+        this.profile.name,
+        this.confirm.name,
+        'name'
+      );
+      this.invalidFields.nameMatch = this.confirmMatchBool(this.confirmErrors.name);
+    },
+    debouncedConfirmEmail: function() {
+      this.confirmErrors.email = this.confirmFields(
+        this.profile.email,
+        this.confirm.email,
+        'email'
+      );
+      this.invalidFields.emailMatch = this.confirmMatchBool(this.confirmErrors.email);
+    },
+    debouncedConfirmPassword: function() {
+      this.confirmErrors.password = this.confirmFields(
+        this.profile.password,
+        this.confirm.password,
+        'password'
+      );
+      this.invalidFields.passwordMatch = this.confirmMatchBool(this.confirmErrors.password);
+    },
+    preview: function(event) {
+      // preview selected image file
+      var input = event.target;
+      if (input.files && input.files[0]) {
+        var reader = new FileReader();
+        reader.onload = e => {
+          this.userImage = e.target.result;
+        };
+        reader.readAsDataURL(input.files[0]);
+      }
+    },
+    getUpdated: function(isEdited, newVal, oldVal) {
+      return isEdited ? newVal : oldVal;
+    },
+    getName: function() {
+      return this.getUpdated(
+        this.editFields.name, 
+        this.profile.name, 
+        this.sessionInfo.name)
+    },
+    getEmail: function() {
+      return this.getUpdated(
+        this.editFields.email, 
+        this.profile.email, 
+        this.sessionInfo.email)
+    },
+    getPassword: function() {
+      return this.getUpdated(
+        this.editFields.password, 
+        this.profile.password, 
+        this.profile.password)
+    },
+    submit: function(evt) {
+      // confirm matches prior to submitting to prevent user
+      // from submitting without typing anything in the confirmation fields
+      this.confirmAllMatches()
+      .then(res => {
+      if (this.checkErrors === false) {
+        localforage.getItem("X_TOKEN")
+        .then(session => {
+          let name = this.getName();
+          let email = this.getEmail();
+          let password = this.getPassword();
+          // axios.post(`/api/v1/profile/edit/${this.sessionInfo._id.$oid}`, {
```

Comment:
If this is unused, let's delete it.

Reply:
This change is implemented in a new branch and PR -- see the branch profile-edits

Reply:
This PR must still be merged, as the changes here are not included in the profile-edits PR